### PR TITLE
Fix Python Failures

### PR DESCRIPTION
**Pull request by TimPansino**
* Fix an issue where init container builds would not properly mark as failed due to an incorrect exit status in minikube.
  * See https://github.com/kubernetes/minikube/issues/16576
  * An example of proper failing behavior after this fix: https://github.com/newrelic/newrelic-agent-init-container/actions/runs/9999193077/job/27639712100?pr=74
* Fix an issue in downloading the exact agent version being the wrong type after unifying extraction logic with the lambda layers build process.

### CODE REVIEW

**Review by lrafeei — APPROVED** [review]

**Comment by TimPansino** [comment]
Woah that's good to know, that doesn't require more than 1 of the code owner teams to review before merging is enabled. The automerge button is a bit more dangerous than I realized 😅

**Pull request merged**
